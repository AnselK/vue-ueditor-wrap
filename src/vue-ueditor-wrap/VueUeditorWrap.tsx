import { PropType, CSSProperties, defineComponent } from 'vue';

export type ButtonType = 'primary' | 'success' | 'warning' | 'danger';

export default defineComponent({
  name: 'VueUeditorWrap',

  props: {
    text: String,
    color: String,
    type: {
      type: String as PropType<ButtonType>,
      default: 'primary',
    },
  },

  emits: ['click'],

  setup(props, { emit, slots }) {
    const onClick = (event: MouseEvent) => {
      emit('click', event);
    };

    const renderText = () => {
      const text = slots.default ? slots.default() : props.text;

      if (text) {
        return <span>{text}</span>;
      }
    };

    const getStyle = () => {
      const { color } = props;
      if (color) {
        const style: CSSProperties = {};
        style.color = color;
        return style;
      }
    };

    const classes = ['demo-button', props.type];

    return () => {
      return (
        <button class={classes} style={getStyle()} onClick={onClick}>
          {renderText()}
        </button>
      );
    };
  },
});
