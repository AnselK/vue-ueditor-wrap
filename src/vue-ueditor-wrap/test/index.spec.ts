import { mount } from '@vue/test-utils';
import VueUeditorWrap from '..';

test('render vue-ueditor-wrap', () => {
  const wrapper = mount(VueUeditorWrap);
  expect(wrapper.html()).toMatchSnapshot();
});
