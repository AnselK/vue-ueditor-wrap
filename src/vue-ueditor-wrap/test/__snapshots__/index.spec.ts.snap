// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render demo button 1`] = `
<button class="demo-button primary">
</button>
`;

exports[`render vue-ueditor-wrap 1`] = `
<div>
  <div>
  </div>
</div>
`;
